src/bin/pg_dump/pg_dump.c

```diff
@@ -2156,6 +2159,7 @@ dumpTableData_insert(Archive *fout, const void *dcontext)
 	 * rather than the uninteresting-to-us value.
 	 */
 	attgenerated = (char *) pg_malloc(tbinfo->numatts * sizeof(char));
+	is_sqlvariant_column = (bool *) pg_malloc0(3 * tbinfo->numatts * sizeof(bool));
```

Comment:
do this allocation inside fixCursorForBbfSqlvariantTableData, declare it as NULL here. That way the same variable can be used for both purpose (is_sqlvariant_column && is_sqlvariant_column[field]) and has_sqlvariant_column is no longer needed